### PR TITLE
Add to cart toast update

### PR DESCRIPTION
**Pull request by Damilare-alesh99**
This branch contains the toast after adding items to cart. 

quick notes
- Toast folder was created in components
- Toast does not have an easy way to customise where it shows up apart from the "bottom-right, bottom-left" options that are given in the documentation

Screenshot 2024-11-15 133155

### CODE REVIEW

File: src/components/toasts/AddToCartToast.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          width={80}
+          height={80}
+          alt="rating star"
+          className="rounded"
+        />
+      </div>
+
+      <div className="flex flex-col space-y-1">
+        <div className="font-semibold">Added to cart</div>
+        <div className="text-gray-700">{productName}</div>
```
Comment (KamsiyonnaObi) [inline src/components/toasts/AddToCartToast.tsx]: Let's add `line-clamp-4` in the className for them products with long fancy names

File: src/components/toasts/AddToCartToast.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  price: number;
+  imagePath: string;
+};
+
+const AddToCartToast = ({
+  productName,
+  price,
+  imagePath,
+}: AddToCartToastProps) => {
+  const router = useRouter();
```
Comment (KamsiyonnaObi) [inline src/components/toasts/AddToCartToast.tsx]: Doesn't seem like you used router in this file. I think it's safe to delete it and its import on Line 4

File: src/features/products/components/AddToCartButton.tsx [hunk 3]
```diff
@@ -52,7 +57,18 @@ const AddToCart = ({
...
+              productName={productName}
+              price={price}
+              imagePath={imagePath} />, {
+              position: 'bottom-right',
+              closeButton : true,
+            });
+          } catch (error){
+            toast.error("Failed to add item to cart");
+          }
+        }
```
Comment (KamsiyonnaObi) [inline src/features/products/components/AddToCartButton.tsx:71]:
```chatGPT suggestion
"use client";

import React from "react";
import { Button } from "@/components/ui/button";
import { useCartStore } from "@/store/useCartStore";
import { toast } from "sonner";
import AddToCartToast from "@/components/toasts/AddToCartToast";

type AddToCartProps = {
  productId: string;
  productName: string;
  price: number;
  imagePath: string;
};

const AddToCart = ({
  productId,
  productName,
  price,
  imagePath,
}: AddToCartProps) => {
  const { qty, onAddCartItem, incQty, decQty } = useCartStore();

  // Cart item preparation
  const prepareCartItem = () => ({
    id: productId,
    name: productName,
    quantity: qty,
    price,
    imagePath,
  });

  // Handle Add to Cart
  const handleAddToCart = () => {
    try {
      const cartItem = prepareCartItem();
      onAddCartItem(cartItem, qty);
      showSuccessToast(productName, price, imagePath);
    } catch (error) {
      showErrorToast();
    }
  };

  // Toast notifications
  const showSuccessToast = (
    name: string,
    price: number,
    image: string
  ) => {
    toast.success(
      <AddToCartToast productName={name} price={price} imagePath={image} />,
      {
        position: "bottom-right",
        closeButton: true,
      }
    );
  };

  const showErrorToast = () => {
    toast.error("Failed to add item to cart", {
      position: "bottom-right",
    });
  };

  return (
    <div className="flex w-full flex-col items-start gap-4">
      {/* Quantity controls */}
      <div className="flex items-center rounded border px-2">
        <Button
          variant="ghost"
          className="px-2 py-1"
          onClick={decQty}
          disabled={qty <= 1}
        >
          -
        </Button>
        <span className="w-auto px-4">{qty}</span>
        <Button
          variant="ghost"
          className="px-2 py-1"
          onClick={incQty}
          disabled={qty >= 10}
        >
          +
        </Button>
      </div>

      {/* Add to Cart button */}
      <div className="flex w-full gap-3">
        <Button onClick={handleAddToCart} disabled={qty < 1}>
          Add To Cart
        </Button>
      </div>
    </div>
  );
};

export default AddToCart;

```

take a look at chatGPT's suggestion on how to make this code more readable.

File: src/features/products/components/AddToCartButton.tsx
Comment (KamsiyonnaObi) [inline src/features/products/components/AddToCartButton.tsx:1]: Take a look at chatGPT's suggestion to make this file more readable
```
"use client";

import React from "react";
import { Button } from "@/components/ui/button";
import { useCartStore } from "@/store/useCartStore";
import { toast } from "sonner";
import AddToCartToast from "@/components/toasts/AddToCartToast";

type AddToCartProps = {
  productId: string;
  productName: string;
  price: number;
  imagePath: string;
};

const AddToCart = ({
  productId,
  productName,
  price,
  imagePath,
}: AddToCartProps) => {
  const { qty, onAddCartItem, incQty, decQty } = useCartStore();

  // Cart item preparation
  const prepareCartItem = () => ({
    id: productId,
    name: productName,
    quantity: qty,
    price,
    imagePath,
  });

  // Handle Add to Cart
  const handleAddToCart = () => {
    try {
      const cartItem = prepareCartItem();
      onAddCartItem(cartItem, qty);
      showSuccessToast(productName, price, imagePath);
    } catch (error) {
      showErrorToast();
    }
  };

  // Toast notifications
  const showSuccessToast = (
    name: string,
    price: number,
    image: string
  ) => {
    toast.success(
      <AddToCartToast productName={name} price={price} imagePath={image} />,
      {
        position: "bottom-right",
        closeButton: true,
      }
    );
  };

  const showErrorToast = () => {
    toast.error("Failed to add item to cart", {
      position: "bottom-right",
    });
  };

  return (
    <div className="flex w-full flex-col items-start gap-4">
      {/* Quantity controls */}
      <div className="flex items-center rounded border px-2">
        <Button
          variant="ghost"
          className="px-2 py-1"
          onClick={decQty}
          disabled={qty <= 1}
        >
          -
        </Button>
        <span className="w-auto px-4">{qty}</span>
        <Button
          variant="ghost"
          className="px-2 py-1"
          onClick={incQty}
          disabled={qty >= 10}
        >
          +
        </Button>
      </div>

      {/* Add to Cart button */}
      <div className="flex w-full gap-3">
        <Button onClick={handleAddToCart} disabled={qty < 1}>
          Add To Cart
        </Button>
      </div>
    </div>
  );
};

export default AddToCart;

```

**Review by KamsiyonnaObi — CHANGES_REQUESTED** [review]
Well done bro, this looks great. Just a few things to look after

File: src/components/toasts/AddToCartToast.tsx (hunk 1 above)
Comment (KamsiyonnaObi) [reply src/components/toasts/AddToCartToast.tsx]: ClassName on line 37

File: src/components/toasts/AddToCartToast.tsx (hunk 2 above)
Comment (Damilare-alesh99) [reply src/components/toasts/AddToCartToast.tsx]: used on line 45

**Pull request merged**